src/Umbraco.Web/Trees/FileSystemTreeController.cs

```diff
@@ -4,14 +4,46 @@
 using System.Net.Http.Formatting;
 using System.Web;
 using Umbraco.Core;
+using Umbraco.Core.Cache;
+using Umbraco.Core.Configuration;
 using Umbraco.Core.IO;
+using Umbraco.Core.Logging;
+using Umbraco.Core.Mapping;
+using Umbraco.Core.Persistence;
+using Umbraco.Core.Services;
 using Umbraco.Web.Actions;
+using Umbraco.Web.Composing;
 using Umbraco.Web.Models.Trees;
+using Umbraco.Web.Routing;
 
 namespace Umbraco.Web.Trees
 {
     public abstract class FileSystemTreeController : TreeController
     {
+        private readonly IMenuItemCollectionFactory _menuItemCollectionFactory;
+
+        protected FileSystemTreeController()
```

Comment:
how come we have this ctor?

Reply:
The controller is abstract, so this ctor is introduced to avoid all inheritors to implement the full ctor. - This is the same way all abstract controllers constructors are handled.